### PR TITLE
Withdrawal persistence

### PR DESCRIPTION
**Pull request by jurgenwerk**
TODO:

- [x] More tests
- [x] Explore the consequences of https://github.com/cardstack/cardstack/pull/2141/commits/6fee41aa9d388ca314cd53d00abe55c1081d9290

### CODE REVIEW

File: packages/web-client/app/utils/web3-strategies/layer2-chain.ts [hunk 1]
```diff
@@ -434,14 +434,15 @@ export default abstract class Layer2ChainWeb3Strategy
...
-    let transactionHash = await new Promise<TransactionHash>((res) => {
-      tokenBridge.relayTokens(
-        safeAddress,
-        tokenAddress,
-        receiverAddress,
-        amountInWei,
-        { onTxnHash: (txnHash) => res(txnHash) }
-      );
+
+    let transactionHash = await new Promise<TransactionHash>((res, reject) => {
```
Comment (jurgenwerk) [inline packages/web-client/app/utils/web3-strategies/layer2-chain.ts:438]: Previously, the result of this didn't actually throw an error, meaning the error handling in the token bridge card in the withdrawal workflow didn't work.

File: packages/web-client/app/components/card-pay/withdrawal-workflow/transaction-amount/index.hbs [hunk 2]
```diff
@@ -47,6 +47,7 @@
             @symbol={{this.currentTokenSymbol}}
             @invalid={{this.isInvalid}}
             @errorMessage={{this.validationMessage}}
+            @disabled={{or @frozen (eq this.amountCtaState 'memorialized')}}
```
Comment (jurgenwerk) [inline packages/web-client/app/components/card-pay/withdrawal-workflow/transaction-amount/index.hbs:50]: Disables the amount when the amount has been used for withdrawal operations later on: 

image

**Comment by jurgenwerk** [comment]
@lukemelia would appreciate it if you can check this even though it's still a draft. I think the only thing missing here are tests, which I'm currently working on. I have some issues understanding how to use the test layer to resolve the async tasks in tests, for example `awaitBridgingTask` and `resumeClaimBridgedTokensTask`. Maybe it would go faster if you have some time for pairing today?

**Comment by lukemelia** [comment]
FYI I rebased this on main now that the workflowSession type safety PR is merged.

**Review by lukemelia — CHANGES_REQUESTED** [review]
The implementation looks good, Matic, and it works great in my testing. Just needs more test coverage.

**Comment by jurgenwerk** [comment]
@lukemelia great, thanks for rebasing! I'll add more test coverage and wrap this up today.

**Comment by jurgenwerk** [comment]
@lukemelia I increased the test coverage with a couple of more acceptance tests. Regarding the change in https://github.com/cardstack/cardstack/commit/6fee41aa9d388ca314cd53d00abe55c1081d9290, I couldn't find any regressions 🤷‍♂️ I think this is now safe to merge, let me know if you have any comments.

**Pull request merged**